spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/Tokenizer.java

```diff
@@ -49,80 +46,83 @@
-        .getSchema().getType();
-      Preconditions.checkArgument(columnToBeTokenizedType == Schema.Type.STRING,
-                                  "Column to be tokenized %s must be of type String, but was of type %s.",
-                                  config.columnToBeTokenized, columnToBeTokenizedType);
+      Schema schema = inputSchema.getField(config.columnToBeTokenized).getSchema();
+      Schema.Type type = schema.isNullable() ? schema.getNonNullable().getType() : schema.getType();
+      Preconditions.checkArgument(type == Schema.Type.STRING, "Column to be tokenized %s must be of type String, " +
+        "but was of type %s.", config.columnToBeTokenized, type);
     }
   }
 
   @Override
+  public void initialize(SparkExecutionPluginContext context) throws Exception {
+    super.initialize(context);
+    tokenizer = new RegexTokenizer().setInputCol(config.columnToBeTokenized).setOutputCol(config.outputColumn)
+      .setPattern(config.patternSeparator);
+  }
+
+  @Override
   public JavaRDD<StructuredRecord> transform(SparkExecutionPluginContext context,
                                              JavaRDD<StructuredRecord> input) throws Exception {
     JavaSparkContext javaSparkContext = context.getSparkContext();
     SQLContext sqlContext = new SQLContext(javaSparkContext);
-    //Create outputschema
-    outputSchema = Schema.recordOf("outputSchema", Schema.Field.of(config.outputColumn,
-                                                                   Schema.arrayOf(Schema.of(Schema.Type.STRING))));
-    //Schema to be used to create dataframe
-    StructType schema = new StructType(new StructField[]{
-      new StructField(config.columnToBeTokenized, DataTypes.StringType, false, Metadata.empty())});
-    //Transform input i.e JavaRDD<StructuredRecord> to JavaRDD<Row>
-    JavaRDD<Row> rowRDD = input.map(new Function<StructuredRecord, Row>() {
+    if (input == null) {
+      return context.getSparkContext().emptyRDD();
+    }
+    outputSchema = outputSchema != null ? outputSchema : config.getOutputSchema(input.first().getSchema(),
+                                                                                config.outputColumn);
+    JavaRDD<String> javardd = input.map(new Function<StructuredRecord, String>() {
       @Override
-      public Row call(StructuredRecord rec) throws Exception {
-        return RowFactory.create(rec.get(config.columnToBeTokenized));
+      public String call(StructuredRecord structuredRecord) throws Exception {
+        return StructuredRecordStringConverter.toJsonString(structuredRecord);
       }
     });
-    DataFrame sentenceDataFrame = sqlContext.createDataFrame(rowRDD, schema);
-    RegexTokenizer tokenizer = new RegexTokenizer().setInputCol(config.columnToBeTokenized)
-      .setOutputCol(config.outputColumn).setPattern(config.pattern);
+    DataFrame sentenceDataFrame = sqlContext.read().json(javardd);
```

Comment:
I see no reason to keep it a SparkCompute, I think we should make it a Transform.